Add documentation for new bloom filter settings (#6449)
opensearch-project/documentation-website: _install-and-configure/configuring-opensearch/index-settings.md

- `index.final_pipeline` (String): The final ingest node pipeline for the index. If the final pipeline is set and the pipeline does not exist, then index requests fail. The pipeline name `_none` specifies that the index does not have an ingest pipeline.

> - `index.optimize_doc_id_lookup.fuzzy_set.enabled` (Boolean): This setting controls whether fuzzy set should be enabled for optimizing document id lookups in indexing/search calls by using an additional data structure (Bloom Filter). Enabling this improves performance for upsert and search operations which rely on document id by creating a new data structure (bloom filter) which allows to handle negative cases (i.e. ids being absent in the existing index) through faster off-heap look-ups. Default is `false`. This setting can only be used if the feature flag `opensearch.experimental.optimize_doc_id_lookup.fuzzy_set.enabled` is set to `true`.

Reviewer: ```suggestion
- `index.optimize_doc_id_lookup.fuzzy_set.enabled` (Boolean): This setting controls whether `fuzzy_set` should be enabled for optimizing document ID lookups in indexing or searching calls by using an additional data structure, in this case, the Bloom filter data structure. Enabling this setting improves performance for upsert and search operations that rely on document ID by creating a new data structure (Bloom filter). The document ID allows for the handling of negative cases (that is, IDs being absent in the existing index) through faster off-heap look-ups. Default is `false`. This setting can only be used if the feature flag `opensearch.experimental.optimize_doc_id_lookup.fuzzy_set.enabled` is set to `true`.
```

Reviewer: Is it the document ID that allows for handling negative cases?

Author: @vagimeli Bloom filter allows for handling negative cases

> - `index.optimize_doc_id_lookup.fuzzy_set.enabled` (Boolean): This setting controls whether `fuzzy_set` should be enabled for optimizing document ID lookups in indexing or searching calls by using an additional data structure, in this case, the Bloom filter data structure. Enabling this setting improves performance for upsert and search operations that rely on document ID by creating a new data structure (Bloom filter). The Bloom filter allows for the handling of negative cases (that is, IDs being absent in the existing index) through faster off-heap look-ups. Default is `false`. This setting can only be used if the feature flag `opensearch.experimental.optimize_doc_id_lookup.fuzzy_set.enabled` is set to `true`.

Reviewer 2: ```suggestion
- `index.optimize_doc_id_lookup.fuzzy_set.enabled` (Boolean): This setting controls whether `fuzzy_set` should be enabled for optimizing document ID lookups in indexing or searching calls by using an additional data structure, in this case, the Bloom filter data structure. Enabling this setting improves performance for upsert and search operations that rely on document ID by creating a new data structure (Bloom filter). The Bloom filter allows for the handling of negative cases (that is, IDs being absent in the existing index) through faster off-heap lookups. Default is `false`. This setting can only be used if the feature flag `opensearch.experimental.optimize_doc_id_lookup.fuzzy_set.enabled` is set to `true`.
```

> - `index.optimize_doc_id_lookup.fuzzy_set.enabled` (Boolean): This setting controls whether `fuzzy_set` should be enabled for optimizing document ID lookups in indexing or searching calls by using an additional data structure, in this case, the Bloom filter data structure. Enabling this setting improves performance for upsert and search operations that rely on document ID by creating a new data structure (Bloom filter). The Bloom filter allows for the handling of negative cases (that is, IDs being absent in the existing index) through faster off-heap look-ups. Default is `false`. This setting can only be used if the feature flag `opensearch.experimental.optimize_doc_id_lookup.fuzzy_set.enabled` is set to `true`.

Reviewer 2: "in order to optimize" instead of "for optimizing"? "lookups in index or search calls"?

> - `index.optimize_doc_id_lookup.fuzzy_set.false_positive_probability` (Double): Set the false positive probability for the underlying fuzzy set (i.e. bloom filter). A lower false positive probability ensures higher throughput improvement for upsert/get operations. Allowed values are in the range `0.01` and `0.50`. Default is `0.20`. This setting can only be used if the feature flag `opensearch.experimental.optimize_doc_id_lookup.fuzzy_set.enabled` is set to `true`.

Reviewer: ```suggestion
- `index.optimize_doc_id_lookup.fuzzy_set.false_positive_probability` (Double): Set the false-positive probability for the underlying `fuzzy_set` (that is, the Bloom filter). A lower false-positive probability ensures higher throughput improvement for `UPSERT` or `GET` operations. Allowed values range between `0.01` and `0.50`. Default is `0.20`. This setting can only be used if the feature flag `opensearch.experimental.optimize_doc_id_lookup.fuzzy_set.enabled` is set to `true`.
```

> - `index.optimize_doc_id_lookup.fuzzy_set.false_positive_probability` (Double): Set the false-positive probability for the underlying `fuzzy_set` (that is, the Bloom filter). A lower false-positive probability ensures higher throughput improvement for `UPSERT` or `GET` operations. Allowed values range between `0.01` and `0.50`. Default is `0.20`. This setting can only be used if the feature flag `opensearch.experimental.optimize_doc_id_lookup.fuzzy_set.enabled` is set to `true`.

Reviewer 2: "Sets"?

> - `index.optimize_doc_id_lookup.fuzzy_set.false_positive_probability` (Double): Set the false-positive probability for the underlying `fuzzy_set` (that is, the Bloom filter). A lower false-positive probability ensures higher throughput improvement for `UPSERT` or `GET` operations. Allowed values range between `0.01` and `0.50`. Default is `0.20`. This setting can only be used if the feature flag `opensearch.experimental.optimize_doc_id_lookup.fuzzy_set.enabled` is set to `true`.

Reviewer 2: ```suggestion
- `index.optimize_doc_id_lookup.fuzzy_set.false_positive_probability` (Double): Set the false-positive probability for the underlying `fuzzy_set` (that is, the Bloom filter). A lower false-positive probability ensures higher throughput for `UPSERT` and `GET` operations. Allowed values range between `0.01` and `0.50`. Default is `0.20`. This setting can only be used if the feature flag `opensearch.experimental.optimize_doc_id_lookup.fuzzy_set.enabled` is set to `true`.
```
